harness/determined/launch/deepspeed.py

```diff
@@ -153,28 +162,53 @@ def create_deepspeed_env_file() -> None:
         environ = os.environ.copy()
         for k, v in environ.items():
             if k in INCLUDE:
-                f.write(f"{k}={v}\n")
+                # We need to turn our envvars into shell-escaped strings to export them correctly
+                # since values may contain spaces and quotes.  shlex.quote was removed from the
+                # deepspeed launcher in 0.6.2 so we add it here for this version onwards.
+                if deepspeed_version >= version.parse("0.6.2"):
+                    f.write(f"{k}={shlex.quote(v)}\n")
+                else:
+                    f.write(f"{k}={v}\n")
 
 
-def create_run_command(master_address: str, hostfile_path: str) -> List[str]:
+def create_run_command(master_address: str, hostfile_path: Optional[str]) -> List[str]:
     # Construct the deepspeed command.
-    deepspeed_process_cmd = [
-        "deepspeed",
-        "-H",
-        hostfile_path,
-        "--master_addr",
-        master_address,
-        "--no_python",
-        "--no_local_rank",
-        "--",
-    ]
+    deepspeed_process_cmd = ["deepspeed"]
+    if hostfile_path is not None:
+        deepspeed_process_cmd += ["-H", hostfile_path]
+    deepspeed_process_cmd += ["--master_addr", master_address, "--no_python", "--no_local_rank"]
+    if deepspeed_version > version.parse("0.6.4"):
+        deepspeed_process_cmd.append("--no_ssh_check")  # Bypass deepspeed's ssh check.
+    deepspeed_process_cmd.append("--")
     return deepspeed_process_cmd
 
 
+def check_deepspeed_version(multi_machine: bool) -> None:
+    if not multi_machine:
+        return
+    # Upstream deepspeed added an ssh check from 0.6.1 onwards that does not have the
+    # StrictHostKeyChecking=no arg and fails with our agents.  A PR adding a no_ssh_check arg
+    # to bypass this should land for versions 0.6.5 and onwards.
```

Comment:
it already did land, right?

Reply:
yes, latest version will work with our code